Controller/HelperController.php

```diff
@@ -147,9 +147,10 @@ public function retrieveFormFieldElementAction(Request $request)
 
         $admin->setSubject($subject);
 
-        $formBuilder = $admin->getFormBuilder($subject);
+        $formBuilder = $admin->getFormBuilder();
 
         $form = $formBuilder->getForm();
+        $form->setData($subject);
```

Comment:
> How/why does this work?

For the same reason https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AdminHelper.php#L112-L115 works :/

`AdminInterface::getFormBuilder` signature doesn't accept any parameter currently (it do nothing with `$subject`) so, before that, the form does not have a binded data (`$subject`) i.e their children cannot access to `$form->getParent()->getData()` into a custom form field type.

This happen when we use `sonata_type_model`.